pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/OffHeapStarTreeNode.java

```diff
@@ -39,11 +39,13 @@ public class OffHeapStarTreeNode implements StarTreeNode {
   private final PinotDataBuffer _dataBuffer;
   private final int _nodeId;
   private final int _firstChildId;
+  private final int _lastChildId;
```

Comment:
Is this change required? This will add extra memory footprint to the star-tree node

Reply:
this can be avoided, by creating a new iterator in the CachedStarTreeNode instead of delegating to OffHeapStarTreeNode's iterator, when it reads lastChildId but that's already too late.

Reply:
Looks like _lastChildId is needed to keep the abstraction clean to delegate calls from CachedStarTreeNode to OffHeapStarTreeNode. This extra int field doesn't add more overhead considering the object memory layout.

Reply:
I’m not sure there is any alignment padding for it to fit into, I measured that here: https://github.com/apache/pinot/pull/8570#issue-1209642509

Reply:
I counted obj ref as 8, but think I should count it as 4 bytes as UseCompressedOops is used in most cases

Reply:
For future reference, in case you haven't used it before, you can actually measure it on Hotspot using [jol](https://mvnrepository.com/artifact/org.openjdk.jol/jol-core/0.16). You just write something like:

```java
System.out.println(ClassLayout.parseClass(StarTreeNode.class).toPrintable());
```

But this takes into account you JVM settings, so if you really want to go above 32GiB heap you'd have to disable compressed oops.